wandb/apis/public/artifacts.py

```diff
@@ -402,6 +403,44 @@ def load(self):
         self._attrs = response["project"]["artifactType"]["artifactCollection"]
         return self._attrs
 
+    def change_collection_type(self, new_collection_type: str) -> None:
+        """Change the collection type of the artifact.
+
+        Arguments:
+            new_collection_type: The new collection type to use, freeform string.
+        """
+        termlog(
+            f"Changing artifact collection type of "
+            f"{self.type} to {new_collection_type}"
+        )
+        template = """
+            mutation MoveArtifactCollection(
+                $artifactSequenceID: ID!
+                $destinationArtifactTypeName: String!
+            ) {
+                moveArtifactSequence(
+                input: {
+                    artifactSequenceID: $artifactSequenceID
+                    destinationArtifactTypeName: $destinationArtifactTypeName
+                }
+                ) {
+                artifactCollection {
+                    id
+                    name
+                    description
+                    __typename
+                }
+                }
+            }
+            """
+        variable_values = {
+            "artifactSequenceID": self.id,
+            "destinationArtifactTypeName": new_collection_type,
+        }
+        mutation = gql(template)
+        self.client.execute(mutation, variable_values=variable_values)
```

Comment:
This query will fail for portfolios, i.e., if `self.is_sequence()` is `False`. So lets add an if condition here to execute the query only if `self.is_sequence()` is `True`, and otherwise raise an error since the input is invalid

```
if self.is_sequence():
  self.client.execute(...)
  self.type = new_collection_type
else:
  raise ValueError("Artifact collection needs to be a sequence")
```